public/app/features/dashboard/api/ResponseTransformers.ts

```diff
@@ -293,14 +299,23 @@ function getElementsFromPanels(panels: Panel[]): [DashboardV2Spec['elements'], D
 function getDefaultDatasourceType() {
   const datasources = config.datasources;
   // find default datasource in datasources
-  return Object.values(datasources).find((ds) => ds.isDefault)!.type;
+  const dsList = Object.values(datasources);
+  let dsType = dsList.find((ds) => ds.isDefault)?.type;
+  if (!dsType) {
+    dsType = getDefaultDataSourceRef()?.type ?? 'grafana';
+  }
+
+  return dsType;
```

Comment:
Isn't this enough? 

```suggestion
return getDefaultDataSourceRef()?.type ?? 'grafana';
```

Reply:
Not really, the `getDefaultDataSourceRef` is only checking this prop `config.bootData.settings.datasources` , and the `dsList` is consuming `config.bootData.datasources`, and within that list we can have the `isDefaultDs` prop

Reply:
yeah, but it will be the same data source, won't it?

Reply:
made the changes :)